src/crypto/browser/jwkverifier.ts

```diff
@@ -0,0 +1,43 @@
+/**
+ * Copyright 2018 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as base64js from 'base64-js';
+import * as TextEncoding from 'text-encoding-shim';
+
+import {CertVerifier, JwkCertificate} from '../certverifier';
+
+export class JwkVerifier implements CertVerifier {
+  verify(pubkey: JwkCertificate, data: string, signature: string):
```

Comment:
Lets make this an async function, and use `await` instead of creating a new Promise and juggling `then` statements